### PR TITLE
virt plugin: get domains using virConnectListAllDomains

### PR DESCRIPTION
**Pull request by dsilakov**
virConnectListAllDomains was introduced in libvirt almost 4 years ago. Rationale for it can be found in the original RFE: https://www.redhat.com/archives/libvir-list/2012-May/msg00976.html

### CODE REVIEW

**Comment by rubenk** [comment]
I'm sorry, but we can't apply this as is. This breaks on versions where this API is not available.

**Comment by dsilakov** [comment]
Ok, I'll take a look if it is possible to add this in a more graceful way.

File: src/virt.c [hunk 1]
```diff
@@ -705,19 +725,24 @@ refresh_lists (void)
...
             const char *name;
             char *xml = NULL;
             xmlDocPtr xml_doc = NULL;
             xmlXPathContextPtr xpath_ctx = NULL;
             xmlXPathObjectPtr xpath_obj = NULL;
 
+#ifdef HAVE_LIST_ALL_DOMAINS
+            virDomainPtr dom = domains[i];
+#else
+            virDomainPtr dom = NULL;
```
Comment (rubenk) [inline src/virt.c]: these assignments are no-ops and immediately overwritten below.

**Comment by rubenk** [comment]
hi @dsilakov, you might want to hold off with this PR for a bit or discuss with @mojaves since work is  ongoing in the virt plugin in #2037 and #2048

**Comment by dsilakov** [comment]
Yeah, I'll take a look at that PRs. At least they also add switch to modern libvirt API.

**Comment by rubenk** [comment]
the other PR's are fixed now, it would be nice if you could rebase this.

**Comment by rubenk** [comment]
Indeed, the test failure is unrelated.

**Pull request merged**